Mathlib/Computability/PDA.lean

```diff
@@ -0,0 +1,80 @@
+/-
+Copyright (c) 2025 Stefan Hetzl. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Tobias Leichtfried, Stefan Hetzl
+-/
+import Mathlib.Computability.Language
+
+/-!
+# Pushdown Automata
+
+This file contains the definition of a Pushdown Automaton (PDA). Pushdown automata recognize
+exactly the context-free languages.
+-/
+
+/-- A PDA is a set of states (`Q`), a tape alphabet (`T`), a stack alphabet (`S`),  an inital state
+  (`initial_state`), a start symbol (`start_symbol`), a set of final states (`final_states`), and a
+  transition function. The transition function is given in two parts: `transition_fun` reads a
+  letter and `transition_fun'` makes an epsilon transition. -/
+structure PDA (Q T S : Type) [Fintype Q] [Fintype T] [Fintype S] where
+  /-- Initial state. -/
+  initial_state : Q
+  /-- Start symbol on the stack. -/
+  start_symbol : S
+  /-- Set of final states. -/
+  final_states : Set Q
+  /-- Transition function reading a letter from T. -/
+  transition_fun : Q → T → S → Set (Q × List S)
+  /-- Epsilon transition function. -/
+  transition_fun' : Q → S → Set (Q × List S)
+  finite (q : Q)(a : T)(Z : S): (transition_fun q a Z).Finite
+  finite' (q : Q)(Z : S): (transition_fun' q Z).Finite
+
+namespace PDA
+
+variable {Q T S : Type} [Fintype Q] [Fintype T] [Fintype S]
+
+/-- A configuration of a PDA is a state (`state`), the remaining input (`input`), and the current
+  stack. -/
+@[ext]
+structure conf (p : PDA Q T S) where
+  /-- Current state. -/
+  state : Q
+  /-- Remaining input. -/
+  input : List T
+  /-- Current stack. -/
+  stack : List S
+
+variable {pda : PDA Q T S}
```

Comment:
I'd rather treat the automaton as data, hence single-letter name.

Reply:
Do I understand you correctly that you suggest to rename `pda` to, e.g., `M`? Is there some mathlib naming convention I am not aware of?

Reply:
https://leanprover-community.github.io/contribute/naming.html